cmd/security-secrets-setup/main.go

```diff
@@ -16,80 +17,111 @@ package main
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
 )
 
-func main() {
-	start := time.Now()
-	var configFile string
+type exit interface {
+	callExit(int)
+}
+
+type exitCode struct{}
+
+type optionDispatcher interface {
+	run(loggingClient logger.LoggingClient) (int, error)
+}
+
+type pkiInitOptionDispatcher struct{}
 
+var exitInstance = newExit()
+var dispatcherInstance = newOptionDispatcher()
+var configFile string
+
+func init() {
+	// define and register command line flags:
 	flag.StringVar(&configFile, "config", "", "specify JSON configuration file: /path/to/file.json")
 	flag.StringVar(&configFile, "c", "", "specify JSON configuration file: /path/to/file.json")
 	flag.Usage = usage.HelpCallbackSecuritySetup
+}
+
+func main() {
+	start := time.Now()
+
 	flag.Parse()
 
-	if configFile == "" {
+	if len(os.Args) < 2 {
+		fmt.Println("Please specify option for security-secrets-setup.")
 		flag.PrintDefaults()
-		os.Exit(1)
+		exitInstance.callExit(1)
+		return
 	}
 
 	setup.Init()
 
-	// Create and initialize the fs environment and global vars for the PKI materials
-	lc := logger.NewClient("security-secrets-setup", setup.Configuration.Logging.EnableRemote,
-		setup.Configuration.Logging.File, setup.Configuration.Writable.LogLevel)
-
-	// Read the Json config file and unmarshall content into struct type X509Config
-	x509config, err := config.NewX509Config(configFile)
+	loggerAbsPath, err := filepath.Abs(setup.Configuration.Logging.File)
 	if err != nil {
-		lc.Error(err.Error())
+		fmt.Printf("Error on finding the absolute path for logging client from configuration Logging.File file %s: %v\n", setup.Configuration.Logging.File, err)
+		exitInstance.callExit(1)
 		return
 	}
 
-	seed, err := setup.NewCertificateSeed(x509config, setup.NewDirectoryHandler(lc))
-	if err != nil {
-		lc.Error(err.Error())
-		return
-	}
+	// Create and initialize the fs environment and global vars for the PKI materials
+	lc := logger.NewClient("security-secrets-setup", setup.Configuration.Logging.EnableRemote,
+		loggerAbsPath, setup.Configuration.Writable.LogLevel)
 
-	rootCA, err := certificates.NewCertificateGenerator(certificates.RootCertificate, seed, certificates.NewFileWriter(), lc)
-	if err != nil {
-		lc.Error(err.Error())
-		return
-	}
+	if configFile == "" {
+		// run with other options for pki-init
+		statusCode, err := dispatcherInstance.run(lc)
 
-	err = rootCA.Generate()
-	if err != nil {
-		lc.Error(err.Error())
-		return
-	}
+		if err != nil {
+			lc.Error(err.Error())
+		}
 
-	tlsCert, err := certificates.NewCertificateGenerator(certificates.TLSCertificate, seed, certificates.NewFileWriter(), lc)
-	if err != nil {
-		lc.Error(err.Error())
+		exitInstance.callExit(statusCode)
 		return
 	}
 
-	tlsCert.Generate()
-	if err != nil {
+	if err := option.GenTLSAssets(configFile, lc); err != nil {
 		lc.Error(err.Error())
 		return
 	}
+
 	lc.Info("PKISetup complete", internal.LogDurationKey, time.Since(start).String())
```

Comment:
change comment to "secrets setup complete"

Reply:
good catch